### ISSUE TITLE
smallrye-reactive-messaging-jms support for Jackson instead of JsonB for (de-)serialization

### ISSUE THREAD

**Issue by coiouhkc** [smallrye/smallrye-reactive-messaging#1296]
Hi all,
i'm trying to integrate smallrye-reactive-messaging-jms with IBM MQ in a Quarkus-based project.
Quarkus allows for selection of either jsonb or jackson as serialization framework and i'm using jackson in the project.
However smallrye-reactive-messaging-jms explicitly requires use of jsonb, otherwise
```
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:540)
        ... 15 more
Caused by: java.lang.RuntimeException: Failed to initialize Arc
        at io.quarkus.arc.Arc.initialize(Arc.java:26)
        at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:24)
        ... 15 more
Caused by: java.lang.NoClassDefFoundError: Ljavax/json/bind/Jsonb;
        at java.base/java.lang.Class.getDeclaredFields0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3061)
        at java.base/java.lang.Class.getDeclaredField(Class.java:2409)
        at io.quarkus.arc.impl.Reflections.findFieldInternal(Reflections.java:64)
        at io.quarkus.arc.impl.Reflections.access$000(Reflections.java:24)
        at io.quarkus.arc.impl.Reflections$1.apply(Reflections.java:33)
        at io.quarkus.arc.impl.Reflections$1.apply(Reflections.java:30)
        at io.quarkus.arc.impl.ComputingCache$1.get(ComputingCache.java:52)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:49)
        at io.quarkus.arc.impl.ComputingCache.getValue(ComputingCache.java:40)
        at io.quarkus.arc.impl.Reflections.findField(Reflections.java:59)
        at io.smallrye.reactive.messaging.jms.JmsConnector_Bean.<init>(JmsConnector_Bean.zig:138)
        at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans5(Default_ComponentsProvider.zig:3043)
        at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:58)
        at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:112)
        at io.quarkus.arc.Arc.initialize(Arc.java:20)
        ... 19 more
Caused by: java.lang.ClassNotFoundException: javax.json.bind.Jsonb
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:428)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:378)
```
 - is there a possibility to make the choice (jsonb vs jackson) configurable?
Thanks!

**Comment by cescoffier** [comment]
I believe it's possible. However note that the JMS connector is not officially supported in Quarkus. I doubt it will work in native.

**Comment by coiouhkc** [comment]
Hi @cescoffier, thanks for the quick reply. 
I'm using java runtime based images for now, so at least short-term lack of native support might be OK - also explains why there's no `io.quarkus:quarkus-smallrye-reactive-messaging-jms` (btw, what's the reason for your doubts?).
What can i do/ how do we proceed further with jackson support? I'd be glad to try working on a PR, if you'd provide me with a starting point (obviously it's could be
```
@Inject
Instance<Jsonb> jsonb;
```
in `JmsConnector`, but what would be the preferred way of choosing between different implementations?)

**Comment by cescoffier** [comment]
We would need to introduce an abstraction and have two implementation, probably in different modules to avoid having to bring Jackson and JSON-B on the classpath.

**Comment by coiouhkc** [comment]
You mean like `smallrye-reactive-messaging-jms-jsonb` and `smallrye-reactive-messaging-jms-jackson` implementing a common (de)serialization interface defined in `smallrye-reactive-messaging-jms`?

**Comment by cescoffier** [comment]
Yes, something like that. Trying to see if we could find an easier way.

**Comment by coiouhkc** [comment]
@cescoffier could you please check the linked (draft) implementation and tell me whether the general direction is OK?

**Comment by coiouhkc** [comment]
> I doubt it will work in native.

Btw, what's the reason behind your doubt?

**Comment by cescoffier** [comment]
We will need to see, but we will have to extend the native compiler configuration to include SPI implementations.

**Comment by cescoffier** [comment]
Was fixed in #1304
